src/cockpit/channels/filesystem.py

```diff
@@ -214,10 +214,10 @@ def do_done(self):
 
             try:
                 os.rename(self._temppath, self._path)
-            except OSError:
+            except OSError as exc:
```

Comment:
You're interested in the case where the destination is not a directory — in that case catch the specific exception and raise `access-denied` for that case.

For the other case, if you want to catch it (which could make sense) then probably you want to raise internal-error with the message attached.

Reply:
Please also add `FileNotFoundError` mapped to `not-found` — this can happen if the directory is deleted while we're in the middle of writing the temporary file.